src/ros/laser_scan_observer.h

```diff
@@ -8,14 +8,26 @@
 #include "../core/sensor_data.h"
 #include "topic_with_transform.h"
 
+/**
+ * \brief Class responsibilities: observes laser scans and odometry;
+ * converts ROS structures to internal representation.
+ */
 class LaserScanObserver : public TopicObserver<sensor_msgs::LaserScan> {
   using ScanPtr = boost::shared_ptr<sensor_msgs::LaserScan>;
 public: //methods
-
+/**
+ * Initializes base laser scan observer.
+ * \param skip_max_vals A flag that indicates accuracy lasr scan's data.
```

Comment:
consider: "Whether scan points that exceed the max reliable scan-specific distance be skipped."